File: spring2013/codeWar/LRS/Clients/PlayerCSharpAI/api/CombatUnit.cs
// Created by Windward Studios, Inc. (www.windward.net). No copyright claimed - do anything you want with this code.

using System.Xml.Linq;

namespace PlayerCSharpAI.api
{
	/// <summary>
	/// Any unit that fires (Robot and Laser) inherits from this class.
	/// </summary>
	public abstract class CombatUnit
	{
		/// <summary>
		/// Create from passed in XML..
		/// </summary>
		/// <param name="element">Initialize with the values in this object.</param>
		protected CombatUnit(XElement element)
		{
			Location = new BoardLocation(element);
		}

		/// <summary>
		/// The location on the map for this unit. Ignored if the unit is dead.
		/// </summary>
		public BoardLocation Location { get; set; }
	}
}

File: spring2013/codeWar/LRS/Clients/PlayerCppAI/CombatUnit.cpp
#include "CombatUnit.h"

namespace PlayerCSharpAI
{
	namespace api
	{

		PlayerCSharpAI::api::BoardLocation* CombatUnit::getLocation() const
		{
			return m_location;
		}

		void CombatUnit::setLocation(BoardLocation* value)
		{
			m_location = value;
		}

		CombatUnit::CombatUnit() : m_location(NULL)
		{

		}

		CombatUnit::CombatUnit(const xml_node& element)
		{
			m_location = new BoardLocation(element);
		}

		CombatUnit::CombatUnit(const CombatUnit& c)
		{
			m_location = new BoardLocation(*c.m_location);
		}

		CombatUnit::~CombatUnit()
		{
			delete m_location;
		}

		CombatUnit& CombatUnit::operator=(const CombatUnit& c)
		{
			m_location = new BoardLocation(*c.m_location);
			return *this;
		}
	}
}
